pipelines/youtube_data/__init__.py

```diff
@@ -0,0 +1,181 @@
+        end_date: last date of published video
+
+    return:
+        video_ids: list of videos ids
+    """
+    video_ids = []
+    next_page_token = None
+
+    while True:
+        response = youtube.search().list(
+            channelId=channel_id,
+            part="id",
+            maxResults=max_results,
+            publishedAfter=start_date,
+            publishedBefore=end_date,
+            page_token=next_page_token
+        ).execute()
+
+        for item in response['items']:
+            if 'videoId' in item['id']:
+                video_id = item['id']['videoId']
+                video_ids.append(video_id)
+        
+        # check if there are next page results
+        if 'nextPageToken' not in response:
+            break
+
+    return video_ids
+
+def _get_video_details(youtube: Any, video_list: List[str], max_results: int) -> Dict[str, str]:
+    """
+    function to retrieve detail information given an video id
+
+    params:
+        youtube: YouTube resource object
+        video_list: list of Youtube video id
+        max_results: How many results displayed each API call
+
+    return:
+        stats_list: Dict. that contains detail information of the videos
+    """
+    stats_list = []
+
+    # can only get as max_results at a time
+    for i in range(0, len(video_list), max_results):
+        response = youtube.videos().list(
+            part="snippet,contentDetails,statistics",
+            id=video_list[i:i+max_results]
+        ).execute()
+
+        for video in response["items"]:
+            title = video["snippet"]["title"]
+            published = video["snippet"]["publishedAt"]
+            description = video["snippet"]["description"]
+            tags = video["snippet"].get("tags", [])
+            tags_count = len(tags) if tags else 0
+            views_count = video["statistics"].get("viewCount", 0)
+            likes_count = video["statistics"].get("likeCount", 0)
+            dislikes_count = video["statistics"].get("dislikeCount", 0)
+            comments_count = video["statistics"].get("commentCount", 0)
+            stats_dict = dict(
+                title=title,
+                published=published, 
+                description=description,
+                tags=tags,
+                tags_count=tags_count,
+                views_count=views_count,
+                likes_count=likes_count,
+                dislikes_count=dislikes_count,
+                comments_count=comments_count
+            )
+            stats_list.append(stats_dict)
+
+    return stats_list
+
+@dlt.source(name="youtube_data")
+def youtube_data_source(
+    channel_names: List[str],
+    start_date: str,
+    end_date: str,
+    max_results: int,
+    api_secrets_key: dlt.secrets.value
+) -> Sequence[DltResource]:
+    
+    # youtube API using YYYY-MM-DDTHH:MM:SSZ format
+    start_date = start_date+'T00:00:00Z'
+    end_date = end_date+'T00:00:00Z'
+    max_results = 50
```

Comment:
Here `max_results` is hard coded to 50, which could be problematic if a different value is required. It's best to remove the line `max_results = 50` and pass it as a parameter instead. Consider using default value in the function definition if needed.